shared/modules/transaction.utils.ts

```diff
@@ -1,36 +1,33 @@
 import { isHexString } from 'ethereumjs-util';
 import { Interface } from '@ethersproject/abi';
 import { abiERC721, abiERC20, abiERC1155 } from '@metamask/metamask-eth-abis';
+import type EthQuery from '@metamask/eth-query';
 import log from 'loglevel';
-import { TransactionType } from '@metamask/transaction-controller';
+import {
+  TransactionMeta,
+  TransactionType,
+} from '@metamask/transaction-controller';
+import { TransactionParams } from '@metamask/transaction-controller/dist/types';
+
 import { AssetType, TokenStandard } from '../constants/transaction';
 import { readAddressAsContract } from './contract-utils';
 import { isEqualCaseInsensitive } from './string-utils';
 
-/**
- * @typedef { 'transfer' | 'approve' | 'setapprovalforall' | 'transferfrom' | 'contractInteraction'| 'simpleSend' } InferrableTransactionTypes
- */
-
-/**
- * @typedef {object} InferTransactionTypeResult
- * @property {InferrableTransactionTypes} type - The type of transaction
- * @property {string} getCodeResponse - The contract code, in hex format if
- *  it exists. '0x0' or '0x' are also indicators of non-existent contract
- *  code
- */
+const INFERRABLE_TRANSACTION_TYPES: TransactionType[] = [
+  TransactionType.tokenMethodApprove,
+  TransactionType.tokenMethodSetApprovalForAll,
+  TransactionType.tokenMethodTransfer,
+  TransactionType.tokenMethodTransferFrom,
+  TransactionType.contractInteraction,
+  TransactionType.simpleSend,
+];
 
-/**
- * @typedef EthersContractCall
- * @type object
- * @property {any[]} args - The args/params to the function call.
- * An array-like object with numerical and string indices.
- * @property {string} name - The name of the function.
- * @property {string} signature - The function signature.
- * @property {string} sighash - The function signature hash.
- * @property {EthersBigNumber} value - The ETH value associated with the call.
- * @property {FunctionFragment} functionFragment - The Ethers function fragment
- * representation of the function.
- */
+type InferTransactionTypeResult = {
+  // The type of transaction
+  type: TransactionType | undefined;
```

Comment:
This could be removed without breaking any tests. See https://github.com/MetaMask/metamask-extension/pull/23276/files#r1519072905

```suggestion
  type: TransactionType;
```

Reply:
[5d8c13](https://github.com/MetaMask/metamask-extension/pull/23276/commits/5d8c1347481a2e3ad2be7f42b3b93e7eb80b9ffe)